### ISSUE TITLE
[FVT]Please make the priority clearly as node name can be the same with template

### ISSUE THREAD

**Issue by caomengmeng** [xcat2/xcat-core#1471]
As the node name can be the same with a template, please make sure the priority. 
It is still the template priority high than the node, we discussed this before and it is the node priority high than template, so how do we final decision?

```
[root@c910f04x27v05 autotest]# lsdef --template cec-template
Object name: cec-template
    groups=cec,all
    hcp=MANDATORY:The hardware control point(HMC) for the node
    hwtype=cec
    mgt=hmc
    mtm=MANDATORY:The machine type and model number of the node
    nodetype=ppc
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    serial=MANDATORY:The serial number of the node
    usercomment=the cec definition template
[root@c910f04x27v05 autotest]# mkdef -t node -o cec-template --template cec-template serial=test mtm=test hcp=test groups=test_template_priority
1 object definitions have been created or modified.
[root@c910f04x27v05 autotest]# lsdef cec-template
Object name: cec-template
    groups=test_template_priority
    hcp=test
    hwtype=cec
    mgt=hmc
    mtm=test
    nodetype=ppc
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    serial=test
    usercomment=the cec definition template
[root@c910f04x27v05 autotest]# mkdef -t node -o auto_test_cec_node --template cec-template serial=test2 mtm=test2 hcp=test2
1 object definitions have been created or modified.
[root@c910f04x27v05 autotest]#
[root@c910f04x27v05 autotest]# lsdef auto_test_cec_node
Object name: auto_test_cec_node
    groups=cec,all
    hcp=test2
    hwtype=cec
    mgt=hmc
    mtm=test2
    nodetype=ppc
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    serial=test2
    usercomment=the cec definition template
```

See another extreme example below:
```
[root@c910f04x27v05 autotest]# mkdef -t node -o switch-template --template hmc-template ip=test username=test password=test
1 object definitions have been created or modified.
[root@c910f04x27v05 autotest]# lsdef
auto_test_cec_node  (node)
c910f04x18v015  (node)
c910f04x27v06  (node)
c910f04x27v07  (node)
cec-template  (node)
switch-10-4-25-1  (node)
switch-10-5-24-1  (node)
switch-template  (node)
[root@c910f04x27v05 autotest]# lsdef switch-template
Object name: switch-template
    groups=hmc
    hwtype=hmc
    ip=test
    mgt=hmc
    nodetype=hmc
    password=test
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    usercomment=the template for hmc definition
    username=test
[root@c910f04x27v05 autotest]# mkdef -t node -o test_priority --template switch-template
1 object definitions have been created or modified.
[root@c910f04x27v05 autotest]# lsdef test_priority
Object name: test_priority
    groups=switch
    mgt=switch
    nodetype=switch
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    usercomment=the switch definition template
```
Test ENV:
```
[root@c910f04x27v05 autotest]# cat /etc/*release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.2 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.2 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.2"
Red Hat Enterprise Linux Server release 7.2 (Maipo)
Red Hat Enterprise Linux Server release 7.2 (Maipo)
[root@c910f04x27v05 autotest]# lsxcatd -a
Version 2.12.1 (git commit 705c119d0f3166f0532cd401475666c3d5e11d38, built Sun Jul  3 06:45:35 EDT 2016)
This is a Management Node
cfgloc=mysql:dbname=xcatdb;host=10.4.27.5|xcatadmin
dbengine=mysql
dbname=xcatdb
dbhost=10.4.27.5
dbadmin=xcatadmin

```

**Comment by immarvin** [comment]
fixed in https://github.com/xcat2/xcat-core/pull/1473

**Comment by caomengmeng** [comment]
Hi @immarvin, it is reproduced, please help to check it.
```
[root@c910f02c04p08 mkdef]# mkdef -t node -o cec-template --template cec-template serial=test mtm=test hcp=test groups=test_template_priority
1 object definitions have been created or modified.
[root@c910f02c04p08 mkdef]# lsdef
c910f02c04p09  (node)
c910f02c04p16  (node)
c910f02fsp04  (node)
c910hmc01  (node)
cec-template  (node)
[root@c910f02c04p08 mkdef]# mkdef -t node -o auto_test_cec_node --template cec-template serial=test2 mtm=test2 hcp=test2
1 object definitions have been created or modified.
[root@c910f02c04p08 mkdef]# lsdef
auto_test_cec_node  (node)
c910f02c04p09  (node)
c910f02c04p16  (node)
c910f02fsp04  (node)
c910hmc01  (node)
cec-template  (node)
[root@c910f02c04p08 mkdef]# lsdef cec-template
Object name: cec-template
    groups=test_template_priority
    hcp=test
    hwtype=cec
    mgt=hmc
    mtm=test
    nodetype=ppc
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    serial=test
    usercomment=the cec definition template
[root@c910f02c04p08 mkdef]# lsdef auto_test_cec_node
Object name: auto_test_cec_node
    groups=cec,all
    hcp=test2
    hwtype=cec
    mgt=hmc
    mtm=test2
    nodetype=ppc
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    serial=test2
    usercomment=the cec definition template
```
Test EVN: with latest daily build
```
[root@c910f02c04p08 mkdef]# lsxcatd -a
Version 2.12.1 (git commit bed9ed1c8ab84f648bc4a339880edb807f3facad, built Tue Jul  5 02:42:30 EDT 2016)
This is a Management Node
dbengine=SQLite
[root@c910f02c04p08 mkdef]# cat /etc/*release
NAME="Red Hat Enterprise Linux Server"
VERSION="7.1 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.1"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.1 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.1:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.1
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.1"
Red Hat Enterprise Linux Server release 7.1 (Maipo)
Red Hat Enterprise Linux Server release 7.1 (Maipo)
```

**Comment by daniceexi** [comment]
@caomengmeng 
@immarvin mentioned it's a mistake of document. The template object should take the priority.

**Comment by whowutwut** [comment]
@daniceexi  Do we need to create another pull request to merge `master -> 2.12`?

**Comment by daniceexi** [comment]
All the changes in the master branch has been merged to the 2.12 branch. They are even now.

**Comment by daniceexi** [comment]
I have verified the fix in the latest build.
